tfprotov5/tf5server/server.go

```diff
@@ -473,98 +412,103 @@ func New(name string, serve tfprotov5.ProviderServer, opts ...ServeOpt) tfplugin
 	}
 	envVar := conf.envVar
 	if envVar == "" {
-		addr, err := tfaddr.ParseRawProviderSourceString(name)
-		if err != nil {
-			log.Printf("[ERROR] Error parsing provider name %q: %s", name, err)
-		} else {
-			envVar = addr.Type
-		}
+		envVar = logging.ProviderLoggerName(name)
 	}
-	envVar = strings.ReplaceAll(envVar, "-", "_")
 	if envVar != "" {
-		options = append(options, tfsdklog.WithLogName(envVar), tflog.WithLevelFromEnv("TF_LOG_PROVIDER", envVar))
+		options = append(options, tfsdklog.WithLogName(envVar), tflog.WithLevelFromEnv(logging.EnvTfLogProvider, envVar))
 	}
 	return &server{
-		downstream:   serve,
-		stopCh:       make(chan struct{}),
-		tflogOpts:    options,
-		tflogSDKOpts: sdkOptions,
-		name:         name,
-		useTFLogSink: conf.useLoggingSink != nil,
-		testHandle:   conf.useLoggingSink,
+		downstream:      serve,
+		stopCh:          make(chan struct{}),
+		tflogOpts:       options,
+		tflogSDKOpts:    sdkOptions,
+		name:            name,
+		useTFLogSink:    conf.useLoggingSink != nil,
+		testHandle:      conf.useLoggingSink,
+		protocolDataDir: os.Getenv(logging.EnvTfLogSdkProtoDataDir),
+		protocolVersion: "6",
```

Comment:
Is this `6` here correct given that it's in the code for a protocol v5 server?

Reply:
Aye, it is not correct. Good eyes. Thank you.

```suggestion
		protocolVersion: "5",
```